### PR TITLE
Remove TokenKind.kind

### PR DESCRIPTION
**Pull request by ahoppen**
This property isn't used anywhere, dates back to the old parser and isn't public. Let's remove it.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
Could we wait until #1200 is merged?

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by kimdv — APPROVED** [review]

**Pull request merged**
